atomicfu/build.gradle

```diff
@@ -153,63 +164,68 @@ dependencies {
     transformer project(":atomicfu-transformer")
 }
 
-// ==== CONFIGURE JS =====
-
-def compileJsLegacy = tasks.hasProperty("compileKotlinJsLegacy")
-        ? compileKotlinJsLegacy
-        : compileKotlinJs
+// TODO: this JS Legacy configuration should be removed after the switch to Kotlin language version 1.9
+// (right now configurations are removed conditionally for testing of Kotlin 1.9.0-dev builds)
+if (isJsLegacyCompilerEnabled) {
 
-tasks.withType(compileJsLegacy.getClass()) {
-    kotlinOptions {
-        moduleKind = "umd"
-        sourceMap = true
-        metaInfo = true
-    }
-}
+    // ==== CONFIGURE JS =====
 
-apply from: file("$rootProject.projectDir/gradle/node-js.gradle")
-apply from: file("$rootProject.projectDir/gradle/publish-npm-js.gradle")
+    def compileJsLegacy = tasks.hasProperty("compileKotlinJsLegacy")
+            ? compileKotlinJsLegacy
+            : compileKotlinJs
 
-// Workaround the problem with Node downloading
-repositories.whenObjectAdded {
-    if (it instanceof IvyArtifactRepository) {
-        metadataSources {
-            artifact()
+    tasks.withType(compileJsLegacy.getClass()) {
```

Comment:
- `compileJsLegacy` should be already a `Class`.
- after `.withType` `.configureEach { ... }` should be used

Reply:
Fixed

Reply:
With gradle 6.8.3 compilation fails if I remove getClass()

Reply:
I mean in the lines above it could be:
```groovy
    def compileJsLegacy = tasks.hasProperty("compileKotlinJsLegacy")
            ? compileKotlinJsLegacy.getClass()
            : compileKotlinJs.getClass()
```